tap_paycor/__init__.py

```diff
@@ -77,56 +90,92 @@ def refresh_token(args):
 
     # write new config to disk
     with open(args.config_path, 'w') as f:
-        f.write(args.config)
+        f.write(json.dumps(args.config, indent=2))
+
 
 def sync(config, STATE, catalog):
     bookmark_property = 'updated_at'
+    employee_ids = None
+    for stream in catalog.get_selected_streams(STATE):
+        singer.write_schema(
+            stream_name=stream.tap_stream_id,
+            schema=stream.schema.to_dict(),
+            key_properties=stream.key_properties,
+        )
+        if stream.tap_stream_id == 'employee_custom_fields':
+            if employee_ids is None:
+                # this must be handled in employees
+                raise ValueError('You must tap employees before employee_custom_fields')
+            for employee_id in employee_ids:
+                get_records(config, stream, employee_id)
+            continue
```

Comment:
I believe this is the only way that I deviated from the @themightychris' instructions. Instead I:

* Do employees first (changes in catalog.json) and then set aside the `employee_ids` (line 120-121 below)
* When processing the `employee_custom_fields`, verify that employee_ids was set (not `None`) and then loop over the `employee_ids`